ios/FluentUI/Navigation/NavigationController.swift

```diff
@@ -131,6 +155,9 @@ open class NavigationController: UINavigationController {
         if viewController is ShyHeaderController {
             return false
         }
+        if prefersShyHeader {
```

Comment:
Not a fan of these API changes as this feels like something we need to handle internally. `ShyHeaderController` is also an internal control and we should probably try to make sure our clients never have to worry about dealing with it. Wouldn't it work to just check if the parent of the nav controller is a DrawerController and omit the shy header in that case?

Reply:
Also I don't see how this fixes the TwoLineTitleView on landscape since `viewControllerNeedsWrapping` is mostly unchanged barring the addition of `prefersShyHeader` check

Reply:
Fair point about `ShyHeaderController` being internal. That being said, I feel like checking specifically for a DrawerController might be *too* specific, although it could be fine for now. I've been at somewhat of a loss for more elegant solutions.

And as far as fixing `TwoLineTitleView` on landscape, this happens because `prefersShyHeader` defaults to true. Of course, whether or not we want to keep this solution is a separate matter.

Reply:
i agree with Lamine. we shouldn't expose this prefershyheader at all. the default behavior should be pre 0.17 version. can we do it more internally like if user passing in something in navigationItem to actuallly utilize twolinetitleview, then we we use shyheaderlogic?

Reply:
It feels weird to have to set two properties of a `UINavigationItem` to enable two-line title views. Doing this in the navigation controller itself might be a bit more natural.